tutorials/publish/tutorial39.m

```diff
@@ -0,0 +1,187 @@
+%% Description
+% This tutorial fits a spinwave model to simulated powder data.
+
+%% Define instrument resolution parameters for MARI
+% This is a simplified parameterisation of the resolution of a
+% Time-of-Flight spectrometer which will be used to broaden the simulated
+% data. These parameters will depend on the instrument used - the instrument
+% contact should be able to provide these.
+
+Ei = 5; % meV
+
+% Resolution (from PyChop)
+eres = @(en) 512.17*sqrt((Ei-en).^3 .* ( 8.26326e-10*(0.169+0.4*(Ei./(Ei-en)).^1.5).^2 + 2.81618e-11*(1.169+0.4*(Ei./(Ei-en)).^1.5).^2));
```

Comment:
So these parameters are actually for MARI with the Gd chopper running at 400Hz for Ei=25 meV - but the parameters for the Q resolution below is actually for MERLIN... sorry I completely messed up these lines in my original examples.

The clue is in the expression `0.169+0.4*(Ei/(Ei-en))` and `L2 = 2.5`; the ToF resolution has a geometrical factor of `(L3 + L2*(ei/ef)**1.5)/L1` where L1 is the moderator-to-sample distance, L2 is the sample-to-detector distance and L3 is the Fermi-chopper-to-sample distance. For both MARI and MERLIN L1=11.8m, L3=1.7m, but MARI has L2 = 4m whilst MERLIN has L2 = 2.5m. I don't know why but in PyChop the distances gets divided by 10 (🤷?). 

I suggest to be consistent to use the following expressions:

```
% Resolution for MARI with Gd chopper running at 150Hz and Ei=5meV (from PyChop)
eres = @(en) 2.1750e-04*sqrt((Ei-en).^3 .* ( (53.31017*(0.168+0.400*(Ei./(Ei-en)).^1.5)).^2 + (19.57320*(1.168+0.400*(Ei./(Ei-en)).^1.5)).^2) );
% Q-resolution (parameters for MARI)
e2k = @(en) sqrt( en .* (2*1.67492728e-27*1.60217653e-22) )./1.05457168e-34./1e10;
L1 = 11.8;  % Moderator to Sample distance in m
L2 = 4.0;   % Sample to detector distance in m
```

I guess it doesn't actually matter what the parameter values are for the demo, but I'd like them to be correct just in case someone copies them verbatim for their own calculations...

Reply:
Sure, for MnF2 the parameters should be:

```
% Resolution for MARI with Gd chopper running at 200Hz and Ei=20meV (from PyChop)
eres = @(en) 2.1750e-04*sqrt((Ei-en).^3 .* ( (32.27217*(0.168+0.400*(Ei./(Ei-en)).^1.5)).^2 + (14.53577*(1.168+0.400*(Ei./(Ei-en)).^1.5)).^2) );
% Q-resolution (parameters for MARI)
e2k = @(en) sqrt( en .* (2*1.67492728e-27*1.60217653e-22) )./1.05457168e-34./1e10;
L1 = 11.8;  % Moderator to Sample distance in m
L2 = 4.0;   % Sample to detector distance in m
```

I think I can come up with a function that would return the approximate resolution for different ISIS instruments as a function of `Ei` and chopper frequency...

Reply:
If such a thing exists already in PyChop it'd be great to have it here, thanks!